src/app/Models/MatchingAlgorithm.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Models;
+
+use App\Traits\Uuids;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Database\Eloquent\SoftDeletes;
+use Illuminate\Support\Str;
+
+abstract class MatchingAlgorithm extends Model
+{
+    use HasFactory, SoftDeletes, Uuids;
+
+    /**
+     * Indicates if the model's ID is auto-incrementing.
+     *
+     * @var bool
+     */
+
+    public $incrementing = false;
+
+    /**
+     * The data type of the auto-incrementing ID.
+     *
+     * @var string
+     */
+
+    protected $keyType = 'string';
+
+    /**
+     * The attributes that aren't mass assignable.
+     *
+     * @var array
+     */
+
+    protected $guarded = [];
+
+    /**
+     * The attributes that should be hidden for arrays.
+     *
+     * @var array
+     */
+    protected $hidden = ['deleted_at', "created_at", "updated_at"];
+
+    /**
+     * Abstract method to compute Levenshtein similarity.
+     *
+     * @param mixed $person1
+     * @param mixed $person2
+     * @return float
+     */
+    abstract protected function computeSimilarity($person1, $person2);
```

Comment:
Make sure the signature matches your functions in the children classes. 

Also you must check that the field importance variable respects this : 0 <= $importance <= 1

(You can add conditions into your function)